api/src/routing.js

```diff
@@ -158,7 +159,8 @@ if (process.argv.slice(2).includes('--allow-cors')) {
 
 app.use((req, res, next) => {
   req.id = uuid.v4();
-  next();
+  req.headers[serverUtils.REQUEST_ID_HEADER] = req.id;
```

Comment:
Is this needed at all? Surely req.id is available everywhere req.headers is?

Reply:
This is needed because we might immediately proxy the request to couchdb with the router proxy thing. and those requests need the header as well.